packages/shared/src/components/modals/PostModal.tsx

```diff
@@ -32,85 +15,43 @@ interface PostModalProps
   isFetchingNextPage?: boolean;
 }
 
+export const postModalOverlayClasses =
+  'post-modal-overlay bg-overlay-quaternary-onion';
+
 export default function PostModal({
-  className,
-  children,
-  onRequestClose,
   id,
+  className,
   isFetchingNextPage,
+  onRequestClose,
   onPreviousPost,
   onNextPost,
   ...props
 }: PostModalProps): ReactElement {
-  const { alerts } = useContext(AlertContext);
-  const { sidebarRendered } = useSidebarRendered();
-  const [position, setPosition] =
-    useState<CSSProperties['position']>('relative');
-  const { tokenRefreshed } = useContext(AuthContext);
-  useResetScrollForResponsiveModal();
-  useHideOnModal(props.isOpen);
-
-  const {
-    data: postById,
+  const { post, isLoading } = usePostById({ id, isFetchingNextPage });
+  const position = usePostNavigationPosition({
     isLoading,
-    isFetched,
-  } = useQuery<PostData>(
-    ['post', id],
-    () => request(`${apiUrl}/graphql`, POST_BY_ID_QUERY, { id }),
-    { enabled: !!id && tokenRefreshed },
-  );
-  useScrollTopOffset(
-    () => document?.getElementById?.('post-modal')?.parentElement,
-    {
-      onOverOffset: () => position !== 'fixed' && setPosition('fixed'),
-      onUnderOffset: () => position !== 'relative' && setPosition('relative'),
-      offset:
-        sidebarRendered && alerts?.filter
-          ? SCROLL_OFFSET + ONBOARDING_OFFSET
-          : SCROLL_OFFSET,
-    },
-  );
-
-  useEffect(() => {
-    if (isLoading) {
-      const modal = document.getElementById('post-modal');
-
-      if (!modal) {
-        return;
-      }
-
-      const parent = modal.parentElement;
-
-      parent?.scrollTo?.(0, 0);
-      setPosition('relative');
-    }
-  }, [isLoading]);
+    isDisplayed: props.isOpen,
+  });
 
   return (
-    <Modal
-      size={Modal.Size.Large}
-      kind={Modal.Kind.FlexibleTop}
-      {...props}
-      portalClassName={styles.postModal}
-      className={classNames(className, 'post-modal focus:outline-none')}
-      overlayClassName="post-modal-overlay bg-overlay-quaternary-onion"
-      onRequestClose={onRequestClose}
-      id="post-modal"
-    >
+    <BasePostModal {...props} onRequestClose={onRequestClose}>
       <PostContent
         position={position}
-        postById={postById}
+        post={post}
         onPreviousPost={onPreviousPost}
         onNextPost={onNextPost}
         inlineActions
         className={{
           container: 'post-content',
+          fixedNavigation: {
+            container: modalSizeToClassName[ModalSize.XLarge],
+          },
```

Comment:
Can this not be part of the modal guidelines rather?

Reply:
Not sure about that. This is highly unlikely any other place will use the post navigation. I mean, even on article page it is not used.

Reply:
Hmm yeah exactly so could we not simply attach the classes directly to that?

Reply:
Ah right. I applied it now to components where we can directly to it. Thank you!